iothub/service/tests/Exceptions/IotHubServiceExceptionTests.cs

```diff
@@ -0,0 +1,132 @@
+
+        [TestMethod]
+        public void IotHubServiceException_ctor_transient_ok()
+        {
+            // arrange - act
+            HttpStatusCode statusCode = HttpStatusCode.ServiceUnavailable;
+
+            var exception = new IotHubServiceException(
+                Message,
+                statusCode,
+                IotHubServiceErrorCode.ThrottlingException);
+
+            // assert
+            exception.StatusCode.Should().Be(statusCode);
+            exception.Message.Should().Be(Message);
+            exception.IsTransient.Should().BeTrue();
+            exception.TrackingId.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void IotHubServiceException_UnknownErrorCode_NotTransient()
+        {
+            // arrange - act
+            HttpStatusCode statusCode = HttpStatusCode.MovedPermanently; // not HttpStatusCode.RequestTimeout
+
+            var exception = new IotHubServiceException(
+                Message,
+                statusCode,
+                IotHubServiceErrorCode.Unknown);
+
+            // assert
+            exception.StatusCode.Should().Be(statusCode);
+            exception.Message.Should().Be(Message);
+            exception.IsTransient.Should().BeFalse();
+            exception.TrackingId.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void IotHubServiceException_UnknownErrorCode_Transient()
+        {
+            // arrange - act
+            HttpStatusCode statusCode = HttpStatusCode.RequestTimeout;
+
+            var exception = new IotHubServiceException(
+                Message,
+                statusCode,
+                IotHubServiceErrorCode.Unknown);
+
+            // assert
+            exception.StatusCode.Should().Be(statusCode);
+            exception.Message.Should().Be(Message);
+            exception.IsTransient.Should().BeTrue();
+            exception.TrackingId.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void IotHubServiceException_GetObjectData_NullInfoThrows()
+        {
+            // arrange - act
+            HttpStatusCode statusCode = HttpStatusCode.RequestTimeout;
+
+            var exception = new IotHubServiceException(
+                Message,
+                statusCode,
+                IotHubServiceErrorCode.Unknown);
+            var sctx = new StreamingContext();
+
+            // act
+            Action act =  () => exception.GetObjectData(null, sctx);
+
+            // assert
+            act.Should().Throw<ArgumentNullException>();
+        }
+
+        [TestMethod]
+        public void IotHubServiceException_GetObjectData_Ok()
+        {
+            // arrange - act
+            HttpStatusCode statusCode = HttpStatusCode.RequestTimeout;
+
+            var exception = new IotHubServiceException(
+                Message,
+                statusCode,
+                IotHubServiceErrorCode.Unknown);
+            var sInfo = new SerializationInfo(GetType(), new FormatterConverter());
```

Comment:
nit: these variable names could be a bit more descriptive IMO, but it's a subjective opinion